torch_geometric/nn/encoding.py

```diff
@@ -94,3 +96,182 @@ def forward(self, x: Tensor) -> Tensor:
 
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}({self.out_channels})'
+
+
+# TODO: Generalize the module when needed
+class _MLPMixer(torch.nn.Module):
+    """1-layer MLP-mixer for GraphMixer.
+
+    Args:
+        num_tokens (int): The number of tokens (patches) in each sample.
+        in_channels (int): Input channels.
+        out_channels (int): Output channels.
+    """
+    def __init__(
+        self,
+        num_tokens: int,
+        in_channels: int,
+        out_channels: int,
+    ) -> None:
+        super().__init__()
+        # token mixing
+        self.token_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.token_lin_1 = torch.nn.Linear(num_tokens, num_tokens // 2)
+        self.token_lin_2 = torch.nn.Linear(num_tokens // 2, num_tokens)
+
+        # channel mixing
+        self.channel_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.channel_lin_1 = torch.nn.Linear(in_channels, 4 * in_channels)
+        self.channel_lin_2 = torch.nn.Linear(4 * in_channels, in_channels)
+
+        # head
+        self.head_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.head_lin = torch.nn.Linear(in_channels, out_channels)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """
+        Args:
+            x (torch.Tensor): Tensor of size ``[N, num_tokens, in_channels]``
+
+        Returns:
+            Tensor of size ``[N, out_channels]``
+        """
+        # token mixing
+        h = self.token_layer_norm(x).mT
```

Comment:
what is mT?

Reply:
It's a transpose op over the last two dims:
```python
x    # [N, num_tokens, in_channels]
x.mT # [N, in_channels, num_tokens]
```
https://pytorch.org/docs/2.0/tensors.html#torch.Tensor.mT

Reply:
I see. Thanks!